packages/shorebird_cli/lib/src/shorebird_engine_config.dart

```diff
@@ -0,0 +1,39 @@
+// A reference to a [EngineConfig] instance.
+import 'package:scoped/scoped.dart';
+
+final engineConfigRef = create(() => const EngineConfig.empty());
+
+// The [EngineConfig] instance available in the current zone.
+EngineConfig get engineConfig => read(engineConfigRef);
+
+class EngineConfig {
+  EngineConfig({
+    this.localEngineSrcPath,
+    this.localEngine,
+    this.localEngineHost,
+  }) {
+    final args = [localEngineSrcPath, localEngine, localEngineHost];
+    final allArgsAreNull = args.every((arg) => arg == null);
+    final allArgsAreNotNull = args.every((arg) => arg != null);
+    if (!allArgsAreNull && !allArgsAreNotNull) {
```

Comment:
Rather than doing this can we just have a two constructors? One that accepts no args and one that requires all of them?

Reply:
I think there will always be the possibility of a runtime error (we can't know ahead of time what arguments the user will provide), but I've updated this to move the check out of EngineConfig and into the arg parsing code so it's a bit cleaner.